app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LecturesByRoom.kt

```diff
@@ -0,0 +1,27 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+import java.util.HashMap
+
+data class LecturesByRoom(val lectures: List<Lecture>): HashMap<Int, List<Lecture>>() {
+
+    init{
+        val maxRoomIndex = lectures.map { it.roomIndex }.max() ?: 0
+        val roomCount = maxRoomIndex + 1
```

Comment:
:mag: If I am not mistaken this assumes that the room indices come in a consecutive order without gaps such as `0, 1, 2, 3`. I cannot tell if this is the case. Given room indices are `17, 20, 23, 9` the subsequent loop would iterate 24 times.

Reply:
If I found it correctly you are referring to [this loop](https://github.com/EventFahrplan/EventFahrplan/blob/0cc8eb570c71d52c0890bf31eeb061ef6b161702/app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java#L308-L315) which uses `MyApp.roomCount`. As you are looping here based on `lectures` I would encourage you to take the opportunity to skip some rounds.

How about:

``` kotlin
val roomIndices = lectures.distinctBy { it.roomIndex }.map { it.roomIndex }
for (roomIndex in roomIndices) { ... }
```

Reply:
Honestly, I find `val maxRoomIndex = lectures.map { it.roomIndex }.max() ?: 0` unnecessary hard to understand. I went into debugging the values to see what actually happens. I personally advocate for readability / ease of understand over optimizing runtime execution. Nevertheless, we would reduce the number of if-checks inside the for-loop by reducing the room indices up front.

Reply:
Changed in d25b62930434d0badafc0eea7ebd1ff919b1f21c.